### ISSUE TITLE
Unable to use aliases in kitchen.yml

### ISSUE THREAD

**Issue by limitusus** [test-kitchen/test-kitchen#1353]
## Description

From v1.20, we can't use aliases in .kitchen.yml that we could use in 1.19.x

I was using `.kitchen.yml` with aliases to simplify the YAML but after I update to test-kitchen v1.20 I face exceptions that looks caused by #1328 

Simple YAML with an alias

```yaml
default_attributes: &da
  xxx: yyy

suites:
  xxx:
    attribtues:
      <<: *da
```

```ruby
#!/usr/bin/env ruby

yaml = IO.read('test.yaml')

# Before 1.20
require 'safe_yaml/load'
p SafeYAML.load(yaml) # => OK

# After 1.20
require 'yaml'
p YAML.safe_load(yaml, [], [], false) # => raise Psych::BadAlias
```

I want test-kitchen to allow aliases in `.kitchen.yml`

```
# Preferred IMO
require 'yaml'
p YAML.safe_load(yaml, [], [], true) # => OK
```

## Kitchen Version

```ruby
$ bundle exec kitchen --version
Test Kitchen version 1.20.0
```

## ChefDK Version

Irrelevant because it happens before chef runs

## Ruby Version

```ruby
$ ruby --version
ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-linux]
```

## Platform Version

AmazonLinux

```
$ uname -r
4.9.43-17.38.amzn1.x86_64
```

## Replication Case

Use aliases in `.kitchen.yaml`, and then run `kitchen list` or other subcommands that loads `.kitchen.yml`

## Kitchen Output

```ruby
$ bundle exec kitchen list
bundler: failed to load command: kitchen (/home/limitusus/cookbooks/xxx/vendor/bundle/bin/kitchen)
Psych::BadAlias: Unknown alias: da
  /home/limitusus/.rbenv/versions/2.4.2/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:402:in `visit_Psych_Nodes_Alias'
  (too long, snip)
  /home/limitusus/.rbenv/versions/2.4.2/lib/ruby/2.4.0/psych/visitors/visitor.rb:6:in `accept'
  /home/limitusus/.rbenv/versions/2.4.2/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:32:in `accept'
  /home/limitusus/.rbenv/versions/2.4.2/lib/ruby/2.4.0/psych.rb:302:in `safe_load'
  /home/limitusus/cookbooks/xxx/vendor/bundle/gems/test-kitchen-1.20.0/lib/kitchen/loader/yaml.rb:325:in `parse_yaml_string'
  /home/limitusus/cookbooks/xxx/vendor/bundle/gems/test-kitchen-1.20.0/lib/kitchen/loader/yaml.rb:129:in `yaml'
  /home/limitusus/cookbooks/xxx/vendor/bundle/gems/test-kitchen-1.20.0/lib/kitchen/loader/yaml.rb:117:in `combined_hash'
  /home/limitusus/cookbooks/xxx/vendor/bundle/gems/test-kitchen-1.20.0/lib/kitchen/loader/yaml.rb:68:in `read'
  (snip)
```

**Comment by cheeseplus** [comment]
Definitely caused by that change, though we've never explicitly supported aliases and this is the first time I've seen anyone using them in the YAML.

**Comment by limitusus** [comment]
Of course I can unpack aliases but YAML file gets fairly big because of some large attributes.
I would be glad if you could support aliases.

**Comment by coderanger** [comment]
If you want to set default attribute values for all suites, you can do it in the `provisioner:` configuration section.

**Comment by cheeseplus** [comment]
Alternatively, kitchen's ability to render ERB in the YAML file itself can be leveraged to avoid duplication.

Simple example: https://github.com/basho-labs/riak-chef-cookbook/blob/master/.kitchen.yml

**Comment by cheeseplus** [comment]
After some discussion I think we're mostly okay with the change, if you'd like to make the PR @limitusus I'd happily review and merge!

**Comment by limitusus** [comment]
I  just made a PR #1359. Please check it.

**Comment by LevonBecker** [comment]
We use yaml anchors/aliases as well. The fetch option to either pull from env var or if not found use a default is very useful for defaulting to a dev environment, but with a script that switches env vars can use in other accounts etc. I'll try Chefdk 2.4.17. 2.5.3 is the one bombing.  BTW ChefDK 3.0.16 doesn't blow up, but not ready for Chef client 14 etc... community cookbooks still working out the kinks.

**Comment by coderanger** [comment]
@LevonBecker The fix for this was released in Test Kitchen 1.21.0, so you'll need a ChefDK with at least that version.
